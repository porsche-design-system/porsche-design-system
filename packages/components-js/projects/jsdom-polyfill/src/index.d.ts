declare module '@porsche-design-system/components-js/jsdom-polyfill';
