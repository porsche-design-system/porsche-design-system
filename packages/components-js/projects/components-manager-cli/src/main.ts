import { build, buildCommand } from './commands/build';
import { help, helpCommand } from './commands/help';
import { getPwcmCliVersion } from './services/config';

export type CommandMap = {
  [commandName: string]: (commandParameters: string[]) => Promise<void>;
};

const commandMap: CommandMap = {
  [buildCommand]: build,
  [helpCommand]: help,
};

export const runCommand = async (params: string[]) => {
  const applicationVersion = await getPwcmCliVersion();
  const [command, ...commandParams] = params;

  console.log(`\nPorsche Web Components Manager (v${applicationVersion})`);
  if (commandMap[command]) {
    return await commandMap[command](commandParams);
  }

  return await commandMap[helpCommand](commandParams);
};
