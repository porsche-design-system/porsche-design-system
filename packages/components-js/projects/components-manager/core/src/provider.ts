/**
 * this functionality is meant to be used by the provider of a web component
 * library to provide a compatible library
 */

export { loadComponentLibrary, setRegisterComponentsCallback } from './services/library-handler';
