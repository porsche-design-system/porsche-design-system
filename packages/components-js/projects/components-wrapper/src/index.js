import { defineCustomElements } from '@porsche-design-system/components/dist/esm/loader';
import { setRegisterComponentsCallback } from '@porsche-design-system/web-components-manager';

setRegisterComponentsCallback(
  (prefix) =>
    defineCustomElements(window, {
      transformTagName: (tagName) => (prefix ? `${prefix}-${tagName}` : tagName),
    }),
  PORSCHE_DESIGN_SYSTEM_VERSION
);
