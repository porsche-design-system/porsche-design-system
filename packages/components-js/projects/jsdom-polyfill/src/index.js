// polyfills for features used by our components but unsupported within jest (jsdom)
require('construct-style-sheets-polyfill');
require('intersection-observer');
require('matchmedia-polyfill');
require('matchmedia-polyfill/matchMedia.addListener');
require('scroll-behavior-polyfill');

if (!navigator.userAgent.includes('Node.js') && !navigator.userAgent.includes('jsdom')) {
  throw new Error('[Porsche Design System] the jsdom-polyfill sub package should only be used in node and jsdom environments');
}

// skip initial stylesheet fetch
window.PDS_SKIP_FETCH = true;

// Since DSR conversion does not work in jsdom render client side markup instead
process.browser = true;

// jsdom polyfill build does not load anything from CDN and also does not use our web components manager
// therefore, we have to setup the document.porscheDesignSystem ourselves here
// 'ROLLUP_REPLACE_VERSION' is replaced during build
document.porscheDesignSystem = {
  cdn: 'https://cdn.ui.porsche.com', // needs to be set because we're not initializing via components-js load() method which would normally set this
  ['ROLLUP_REPLACE_VERSION']: {
    readyResolve: () => {},
    isReady: () => Promise.resolve(),
  },
};

// workaround for WebComponentManager not working in jsdom because of missing <script type="module"> support
// see: https://github.com/jsdom/jsdom/issues/2475
require('./lib/loader.cjs').defineCustomElements(); // this is executed synchronous since stencil v4

window.PDS_SKIP_FETCH = false;
