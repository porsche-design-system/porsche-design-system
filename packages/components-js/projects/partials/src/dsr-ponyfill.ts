import { hydrateShadowRoots } from '@webcomponents/template-shadowroot';

// biome-ignore lint/suspicious/noPrototypeBuiltins: ok
if (!HTMLTemplateElement.prototype.hasOwnProperty('shadowRoot')) {
  hydrateShadowRoots(document.body);
}
