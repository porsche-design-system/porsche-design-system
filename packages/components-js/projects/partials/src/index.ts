export * from './lib/partials';
export type { Cdn, Format, FormatWithCSP } from './shared';
