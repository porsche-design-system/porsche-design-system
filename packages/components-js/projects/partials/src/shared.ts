export type Cdn = 'auto' | 'cn';

export type Format = 'html' | 'jsx';

const isBrowser = typeof window !== 'undefined' && typeof window.document !== 'undefined';
export const throwIfRunInBrowser = (partialName: string): void => {
  if (isBrowser) {
    throw new Error(
      `Partials should only be used during build time. It looks like you are using '${partialName}' at run time in a browser.`
    );
  }
};
