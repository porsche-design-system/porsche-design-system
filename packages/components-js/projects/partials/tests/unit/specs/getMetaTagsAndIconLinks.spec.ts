import { getMetaTagsAndIconLinks } from '../../../src';
import { render } from '@testing-library/react';

const hash = '[a-z0-9]{32}';
const baseHrefCom = 'https:\\/\\/cdn\\.ui\\.porsche\\.com\\/porsche-design-system\\/meta-icons';
const baseHrefCn = 'https:\\/\\/cdn\\.ui\\.porsche\\.cn\\/porsche-design-system\\/meta-icons';

jest.mock('../../../src/shared');

describe('validation', () => {
  it('should throw error on unprovided appTitle', () => {
    expect(() => getMetaTagsAndIconLinks()).toThrowErrorMatchingInlineSnapshot(
      `"Option "appTitle" is required to output "<meta name="apple-mobile-web-app-title" content="appTitle" />"`
    );
  });
});

describe('format: html', () => {
  it('should return meta tags and icon links', () => {
    const result: string = getMetaTagsAndIconLinks({ appTitle: 'Porsche UX' });
    const regex = new RegExp(
      `^<meta name=theme-color content=#FFF media=\\(prefers-color-scheme:light\\)><meta name=theme-color content=#0E1418 media=\\(prefers-color-scheme:dark\\)><meta name=apple-mobile-web-app-capable content=yes><meta name=apple-mobile-web-app-status-bar-style content=white><meta name=apple-mobile-web-app-title content="Porsche UX"><meta name=msapplication-TileImage content=${baseHrefCom}\/mstile-270x270\\.${hash}\\.png><meta name=msapplication-TileColor content=#FFF><link rel=icon type=image\/png sizes=32x32 href=${baseHrefCom}\/favicon-32x32\\.${hash}\\.png><link rel=apple-touch-icon href=${baseHrefCom}\/apple-touch-icon-180x180\\.${hash}\\.png><link rel=manifest href=${baseHrefCom}\/manifest\\.${hash}\\.webmanifest>$`
    );
    expect(result).toMatch(regex);
  });

  it('should return meta tags and icon links for china cdn', () => {
    const result: string = getMetaTagsAndIconLinks({ appTitle: 'Porsche UX', cdn: 'cn' });
    const regex = new RegExp(
      `^<meta name=theme-color content=#FFF media=\\(prefers-color-scheme:light\\)><meta name=theme-color content=#0E1418 media=\\(prefers-color-scheme:dark\\)><meta name=apple-mobile-web-app-capable content=yes><meta name=apple-mobile-web-app-status-bar-style content=white><meta name=apple-mobile-web-app-title content="Porsche UX"><meta name=msapplication-TileImage content=${baseHrefCn}\/mstile-270x270\\.${hash}\\.png><meta name=msapplication-TileColor content=#FFF><link rel=icon type=image\/png sizes=32x32 href=${baseHrefCn}\/favicon-32x32\\.${hash}\\.png><link rel=apple-touch-icon href=${baseHrefCn}\/apple-touch-icon-180x180\\.${hash}\\.png><link rel=manifest href=${baseHrefCn}\/manifest\\.cn\\.${hash}\\.webmanifest>$`
    );
    expect(result).toMatch(regex);
  });
});

describe('format: jsx', () => {
  it('should return meta tags and icon links', () => {
    const result: JSX.Element = getMetaTagsAndIconLinks({ format: 'jsx', appTitle: 'Porsche UX Platform' });
    const { container } = render(result);
    const regex = new RegExp(
      `^<meta name="theme-color" content="#FFF" media="\\(prefers-color-scheme:light\\)"><meta name="theme-color" content="#0E1418" media="\\(prefers-color-scheme:dark\\)"><meta name="apple-mobile-web-app-capable" content="yes"><meta name="apple-mobile-web-app-status-bar-style" content="white"><meta name="apple-mobile-web-app-title" content="Porsche UX Platform"><meta name="msapplication-TileImage" content="${baseHrefCom}\/mstile-270x270\\.${hash}\\.png"><meta name="msapplication-TileColor" content="#FFF"><link rel="icon" type="image\/png" sizes="32x32" href="${baseHrefCom}\/favicon-32x32\\.${hash}\\.png"><link rel="apple-touch-icon" href="${baseHrefCom}\/apple-touch-icon-180x180\\.${hash}\\.png"><link rel="manifest" href="${baseHrefCom}\/manifest\\.${hash}\\.webmanifest">$`
    );
    expect(container.innerHTML).toMatch(regex);
  });

  it('should return meta tags and icon links for china cdn', () => {
    const result: JSX.Element = getMetaTagsAndIconLinks({ format: 'jsx', appTitle: 'Porsche UX Platform', cdn: 'cn' });
    const { container } = render(result);
    const regex = new RegExp(
      `^<meta name="theme-color" content="#FFF" media="\\(prefers-color-scheme:light\\)"><meta name="theme-color" content="#0E1418" media="\\(prefers-color-scheme:dark\\)"><meta name="apple-mobile-web-app-capable" content="yes"><meta name="apple-mobile-web-app-status-bar-style" content="white"><meta name="apple-mobile-web-app-title" content="Porsche UX Platform"><meta name="msapplication-TileImage" content="${baseHrefCn}\/mstile-270x270\\.${hash}\\.png"><meta name="msapplication-TileColor" content="#FFF"><link rel="icon" type="image\/png" sizes="32x32" href="${baseHrefCn}\/favicon-32x32\\.${hash}\\.png"><link rel="apple-touch-icon" href="${baseHrefCn}\/apple-touch-icon-180x180\\.${hash}\\.png"><link rel="manifest" href="${baseHrefCn}\/manifest\\.cn\\.${hash}\\.webmanifest">$`
    );
    expect(container.innerHTML).toMatch(regex);
  });
});
