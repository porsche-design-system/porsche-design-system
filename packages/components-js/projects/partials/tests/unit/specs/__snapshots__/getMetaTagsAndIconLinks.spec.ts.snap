// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getMetaTagsAndIconLinks() should return valid jsx meta tags and icon links 1`] = `
<meta
  content="#FFFFFF"
  name="theme-color"
/>
`;

exports[`getMetaTagsAndIconLinks() should return valid jsx meta tags and icon links 2`] = `
<meta
  content="yes"
  name="apple-mobile-web-app-capable"
/>
`;

exports[`getMetaTagsAndIconLinks() should return valid jsx meta tags and icon links 3`] = `
<meta
  content="white"
  name="apple-mobile-web-app-status-bar-style"
/>
`;

exports[`getMetaTagsAndIconLinks() should return valid jsx meta tags and icon links 4`] = `
<meta
  content="Porsche UX"
  name="apple-mobile-web-app-title"
/>
`;

exports[`getMetaTagsAndIconLinks() should return valid jsx meta tags and icon links 5`] = `
<meta
  content="https://cdn.ui.porsche.com"
  name="msapplication-TileImage"
/>
`;

exports[`getMetaTagsAndIconLinks() should return valid jsx meta tags and icon links 6`] = `
<meta
  content="#FFFFFF"
  name="msapplication-TileColor"
/>
`;

exports[`getMetaTagsAndIconLinks() should return valid jsx meta tags and icon links 7`] = `
<link
  href="https://cdn.ui.porsche.com"
  rel="icon"
  sizes="32x32"
  type="image/png"
/>
`;

exports[`getMetaTagsAndIconLinks() should return valid jsx meta tags and icon links 8`] = `
<link
  href="https://cdn.ui.porsche.com"
  rel="apple-touch-icon"
/>
`;

exports[`getMetaTagsAndIconLinks() should return valid jsx meta tags and icon links 9`] = `
<link
  href="https://cdn.ui.porsche.com"
  rel="manifest"
/>
`;
